File: transport_marchandises_web/src/models/Vehicule.js
export class Vehicule {
  constructor({
    id = null,
    marque = "",
    modele = "",
    immatriculation = "",
    capacite = 0,
    tarif_km = 0,
  } = {}) {
    this.id = id;
    this.marque = marque;
    this.modele = modele;
    this.immatriculation = immatriculation;
    this.capacite = capacite;
    this.tarif_km = tarif_km;
  }

  /** Convertit l'objet en JSON prêt pour une requête API */
  toJSON() {
    return {
      id: this.id,
      marque: this.marque,
      modele: this.modele,
      immatriculation: this.immatriculation,
      capacite: this.capacite,
      tarif_km: this.tarif_km,
    };
  }

  /** Crée une instance Vehicule à partir du JSON retourné par l'API */
  static fromJSON(data) {
    if (!data) return null;
    return new Vehicule({
      id: data.id,
      marque: data.marque,
      modele: data.modele,
      immatriculation: data.immatriculation,
      capacite: parseFloat(data.capacite),
      tarif_km: parseFloat(data.tarif_km),
    });
  }
}

File: transport_marchandises/lib/models/vehicule.dart
class Vehicules {
  int? id;
  String? marque;
  String? modele;
  String? immatriculation;
  double? capacite;
  double? tarif_km;
  Vehicules({
    this.id,
    this.marque,
    this.modele,
    this.immatriculation,
    this.capacite,
    this.tarif_km,
  });
  Map<String, dynamic> toJSON() {
    return {
      'id': id,
      'marque': marque,
      'modele': modele,
      'immatriculation': immatriculation,
      'capacite': capacite,
      'tarif_km': tarif_km,
    };
  }

  static Vehicules? jsonToVehicule(data) {
    if (data == null) return null;

    return Vehicules(
      id: data["id"],
      marque: data['marque'] as String?,
      modele: data['modele'] as String?,
      immatriculation: data['immatriculation'] as String?,
      capacite: double.parse(data['capacite']) as double?,
      tarif_km: double.parse(data['tarif_km']) as double?,
    );
  }
}
